apps/gitness/src/pages/pull-request/context/pull-request-data-provider.tsx

```diff
@@ -1,127 +1,45 @@
-import { PullRequestState } from '../types/types'
+import useSpaceSSE from '../../../framework/hooks/useSpaceSSE'
 import { extractSpecificViolations } from '../utils'
-import useGetPullRequestTab, { PullRequestTab } from '../../../hooks/useGetPullRequestTab'
-
-export const codeOwnersNotFoundMessage = 'CODEOWNERS file not found'
-export const codeOwnersNotFoundMessage2 = `path "CODEOWNERS" not found`
-export const codeOwnersNotFoundMessage3 = `failed to find node 'CODEOWNERS' in 'main': failed to get tree node: failed to ls file: path "CODEOWNERS" not found`
-export const oldCommitRefetchRequired = 'A newer commit is available. Only the latest commit can be merged.'
-export const prMergedRefetchRequired = 'Pull request already merged'
-export const POLLING_INTERVAL = 10000
-
-interface PullReqChecksDecisionProps {
-  overallStatus: ExecutionState | undefined
-  count: {
-    error: number
-    failure: number
-    pending: number
-    running: number
-    success: number
-    skipped: number
-    killed: number
-  }
-  error: unknown
-  data: TypesPullReqChecks | undefined
-  color: string
-  background: string
-  message: string
-  summaryText: string
-  checkInfo: {
-    title: string
-    content: string
-    color: string
-    status: string
-  }
-}
-
-interface PullRequestDataContextProps {
-  refetchPullReq: () => void
-  refetchActivities: () => void
-  pullReqMetadata: TypesPullReq | undefined
-  repoMetadata: RepoRepositoryOutput | undefined
-  // Add other necessary properties here
-  loading: boolean
-
-  setRuleViolationArr: (arr: { data: { rule_violations: TypesRuleViolations[] } } | undefined) => void
-  pullReqChecksDecision: PullReqChecksDecisionProps
-  prPanelData: {
-    conflictingFiles: string[] | undefined
-    requiresCommentApproval: boolean
-    atLeastOneReviewerRule: boolean
-    reqCodeOwnerApproval: boolean
-    minApproval: number
-    reqCodeOwnerLatestApproval: boolean
-    minReqLatestApproval: number
-    resolvedCommentArr?: { params: number[] }
-    PRStateLoading: boolean
-    ruleViolation: boolean
-    commentsLoading: boolean
-    commentsInfoData: { header: string; content?: string | undefined; status: string }
-    ruleViolationArr:
-      | {
-          data: {
-            rule_violations: TypesRuleViolations[]
-          }
-        }
-      | undefined
-  }
-}
-
-const PullRequestDataContext = createContext<PullRequestDataContextProps | undefined>(undefined)
-
-export const usePullRequestData = () => {
-  const context = useContext(PullRequestDataContext)
-  if (!context) {
-    throw new Error('usePullRequestData must be used within a PullRequestDataProvider')
-  }
-  return context
-}
-
-interface PullRequestDataProviderProps {
-  children: ReactNode
-}
+import { isEqual } from 'lodash-es'
 
-const PullRequestDataProvider: React.FC<PullRequestDataProviderProps> = ({ children }) => {
+const PullRequestDataProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
Instead of stating what `children` are, use `PropsWithChildren` to get the official type
```suggestion
const PullRequestDataProvider: React.FC<PropsWithChildren<unknown>> = ({ children }) => {
```

Reply:
done